src/borg/fuse.py

```diff
@@ -242,16 +242,15 @@ def _create_filesystem(self):
             self.process_archive(self.args.location.archive)
         else:
             self.versions_index = FuseVersionsIndex()
-            archive_names = (x.name for x in self.manifest.archives.list_considering(self.args))
-            for archive_name in archive_names:
+            for archive in self.manifest.archives.list_considering(self.args):
                 if self.versions:
                     # process archives immediately
-                    self.process_archive(archive_name)
+                    self.process_archive(archive.name)
                 else:
                     # lazily load archives, create archive placeholder inode
-                    archive_inode = self._create_dir(parent=1)
-                    self.contents[1][os.fsencode(archive_name)] = archive_inode
-                    self.pending_archives[archive_inode] = archive_name
+                    archive_inode = self._create_dir(parent=1, mtime=int(archive.ts.timestamp() * 1e9))
```

Comment:
move the computation into _create_dir, give the float timestamp as param (or rename mtime param to mtime_ns and give the ns integer value).

there, you need to consider the value of have_fuse_xtime_ns to compute the correct value and choose the correct attribute name to assign that value. see the other place where have_fuse_xtime_ns is used.